piv/src/main/java/com/yubico/yubikit/piv/PivSession.java

```diff
@@ -873,8 +884,13 @@ public void checkKeySupport(KeyType keyType, PinPolicy pinPolicy, TouchPolicy to
         }
 
         // ROCA
-        if (generate && keyType.params.algorithm == KeyType.Algorithm.RSA) {
-            require(FEATURE_RSA_GENERATION);
+        if (keyType.params.algorithm == KeyType.Algorithm.RSA) {
+            if (generate) {
+                require(FEATURE_RSA_GENERATION);
+            }
+            if (keyType.params.bitLength == 3072 || keyType.params.bitLength == 4096) {
```

Comment:
I assume this is what you meant by [this](https://github.com/Yubico/yubikit-android/pull/132/files/76fc590f79b16887d5150157a0c2dd333ff926ca#r1609381215), @dainnilsson & @AdamVe